### PR TITLE
Fix #8525 #8309 Bulk Action button missing and delete button showing for users with no delete access

### LINKED ISSUE #8525: Bulk Action button is missing in list view after ACL
#### Issue

When i make restriction to user by using ACL, There is no "Bulk Action" button in list view.

Screenshot_1

#### Expected Behavior

It should be there like this;
Screenshot_2

#### Steps to Reproduce

1. Make ACL for user (restrict user for edit,delete,export,update for Leads module)
2.  Open Leads list view
3. Select any records
4. See, there is no "Bulk Action"

#### Your Environment

* SuiteCRM Version used: 7.10.22
* Browser name and version (e.g. Chrome Version 51.0.2704.63 (64-bit)): Firefox 72.0.2 (64-bit)
* Environment name and version (e.g. MySQL, PHP 7): Php 7
* Operating System and version (e.g Ubuntu 16.04): Ubuntu 16.04

### PR DESCRIPTION
**Pull request by ojs87**
Similar to proposed fix https://github.com/salesagility/SuiteCRM/pull/8600 . Please close this one if you want me to continue dialogue in the other thread.

Fix https://github.com/salesagility/SuiteCRM/issues/8525 & https://github.com/salesagility/SuiteCRM/issues/8309

## Description
Bulk Action button is not showing for users without delete access. Delete button is still showing for users without delete access.

## Motivation and Context
Provide continuity between different roles in the list view and fix for what I assume was the original design intent.

## How To Test This
Login with a user without delete access for a module and go to the list view of that module. Check Bulk Action menu, notice that it shows 'email' rather than Bulk Action.

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Final checklist

- [x] My code follows the code style of this project found [here](https://docs.suitecrm.com/community/contributing-code/coding-standards/).
- [ ] My change requires a change to the documentation.
- [x] I have read the [**How to Contribute**](https://docs.suitecrm.com/community/contributing-code/) guidelines.

### CODE REVIEW

**Review by jack7anderson7 — APPROVED** [review]

**Comment by jack7anderson7** [comment]
Hi @ojs87,

This PR is fine to continue the thread on. 
Thanks

File: include/ListView/ListViewDisplay.php [hunk 1]
```diff
@@ -363,6 +363,8 @@ protected function buildActionsLink($id = 'actions_link', $location = 'top')
                 } else {
                     $menuItems[] = $this->buildBulkActionButton($location);
                 }
+            } else {
+                $menuItems[] = $this->buildBulkActionButton($location);
             }
```
Comment (clemente-raposo) [inline include/ListView/ListViewDisplay.php:368]: @jack7anderson7 I think we can remove the full ` if and else` and just leave `$menuItems[] = $this->buildBulkActionButton($location);` as done on https://github.com/salesagility/SuiteCRM/pull/8600. `show_action_dropdown_as_delete` Does not seem to be set anywhere on the full codebase apart from the property initialization on this class.

File: include/ListView/ListViewDisplay.php [hunk 2]
```diff
@@ -414,7 +416,8 @@ protected function buildActionsLink($id = 'actions_link', $location = 'top')
 
 
             if (
```
Comment (clemente-raposo) [inline include/ListView/ListViewDisplay.php:418]: @jack7anderson7  Here we should also stop relying on `& !$this->show_action_dropdown_as_delete` for the reason metioned in https://github.com/salesagility/SuiteCRM/pull/9420/files#r785936264

File: include/ListView/ListViewDisplay.php (hunk 1 above)
Comment (clemente-raposo) [reply include/ListView/ListViewDisplay.php:368]: @jack7anderson7 plus, this also seems to require a css fix. See following screenshot of the dropdown after applying the fix.
legacy-dropdown-acls-css-fix

legacy-dropdown-acls-css-fix-dev-tools

legacy-dropdown-acls-css-fix-dev-tools-fixed

Comment (ojs87) [reply include/ListView/ListViewDisplay.php]: Added an empty link item when there's no delete button, The other option is to split Print as PDF, Export and Map into seperate list items which requires a lot of changes to custom ListViewSmarty classes, or create custom css for the list.

Is deleting the show_action_dropdown_as_delete property ok to do?

**Comment by clemente-raposo** [comment]
@ojs87 small thing. could you squash the last 2 commits together please? and add the `Fix #....` to the commit message?

**Review by jack7anderson7 — APPROVED** [review]

**Review by clemente-raposo — APPROVED** [review]
changes look ok

**Pull request merged**